### DISCUSSION TITLE
import.meta.url is undefined at Vercel Edge Runtime

### DISCUSSION THREAD

**Discussion by PonomareVlad** [vercel/community#3316] [Bug Report]
### Page to Investigate

https://vercel.com/docs/concepts/functions/edge-functions/limitations#many-node.js-apis-are-not-available

### Steps to Reproduce

In section [Many Node.js APIs are not available](https://vercel.com/docs/concepts/functions/edge-functions/limitations#many-node.js-apis-are-not-available) we see this point:
- Use ES modules

But ESM feature `import.meta.url` seems not supported by Vercel Edge Runtime.

**Current behavior**

```js
// api/index.mjs

console.log(import.meta) // {}

console.log(import.meta.url) // undefined
```

At Vercel Edge Runtime `import.meta.url` is `undefined`.

**Expected behavior/code**

Expected specification-driven behavior:

```js
// api/index.mjs

console.log(import.meta) // {url: ".../api/index.mjs"}

console.log(import.meta.url) // ".../api/index.mjs"
```

[Comment from Edge Runtime team](https://github.com/vercel/edge-runtime/issues/486#issuecomment-1643455912)

[Repository to reproduce](https://github.com/PonomareVlad/VercelEdgeURL/tree/main)

**Comment by digitaltopo** [comment]
Seeing the same thing. Trying to import a local file as per https://vercel.com/docs/functions/og-image-generation/og-image-examples#using-a-local-image but edge function fails whenever I try to access `import.meta...`
